iroh-store/src/store.rs

```diff
@@ -314,11 +303,9 @@ impl Store {
         Ok(ids)
     }
 
-    #[tracing::instrument(skip(self))]
-    fn next_id(&self) -> u64 {
-        let mut id = self.inner.next_id.write().unwrap();
-        let next_id = *id;
-        *id = id.checked_add(1).expect("store is full");
+    #[tracing::instrument(skip(self, metadata_table))]
+    fn next_id(&self, metadata_table: &Table<u64, &[u8]>) -> u64 {
+        let next_id = metadata_table.iter().unwrap().next_back().map(|(id, _)| id.value()).unwrap_or_default() + 1;
```

Comment:
- are you sure these are sorted? this is why I was using `max` when reading this out
- this seems potentially expensive, did you check that?